util/rowcodec/decoder.go

```diff
@@ -272,10 +272,19 @@ func (decoder *ChunkDecoder) tryAppendHandleColumn(colIdx int, col *ColInfo, han
 	if handle == nil {
 		return false
 	}
-	if col.ID == decoder.handleColIDs[0] {
+	if handle.IsInt() && col.ID == decoder.handleColIDs[0] {
 		chk.AppendInt64(colIdx, handle.IntValue())
 		return true
 	}
+	for i, id := range decoder.handleColIDs {
+		if col.ID == id {
+			_, val, err := codec.DecodeOne(handle.EncodedCol(i))
```

Comment:
We need to use the `ColInfo` to decode the column,

Call `decodeColToChunk`.

Reply:
Need to call this one https://github.com/pingcap/tidb/blob/master/util/codec/codec.go#L972